src/lib/optimizer/strategy/predicate_placement_rule.cpp

```diff
@@ -88,10 +95,118 @@ void PredicatePlacementRule::_push_down_traversal(const std::shared_ptr<Abstract
+              // Input
+              // (l1 AND r2) OR (l2 AND r1)
+              // ^^^^^^^^^^^    ^^^^^^^^^^^ outer_disjunction holds two (or more) elements from flattening the OR.
+              //                            One of these elements is called expression_in_disjunction.
+              //  ^^     ^^                 inner_conjunction holds two (or more) elements from flattening the AND.
+              //                            One of these elements is called expression_in_conjunction.
+
+              std::vector<std::shared_ptr<AbstractExpression>> left_disjunction{};
+              std::vector<std::shared_ptr<AbstractExpression>> right_disjunction{};
+
+              // Tracks whether we had to abort the search for one of the sides as an inner_conjunction was found that
+              // did not cover the side.
+              auto aborted_left_side = false;
+              auto aborted_right_side = false;
+
+              const auto outer_disjunction =
+                  flatten_logical_expressions(push_down_node->predicate(), LogicalOperator::Or);
+              for (const auto& expression_in_disjunction : outer_disjunction) {
+                // For the current expression_in_disjunction, these hold the PredicateExpressions that need to be true
+                // on the left/right side
+                std::vector<std::shared_ptr<AbstractExpression>> left_conjunction{};
+                std::vector<std::shared_ptr<AbstractExpression>> right_conjunction{};
+
+                // Fill left/right_conjunction
+                const auto inner_conjunction =
+                    flatten_logical_expressions(expression_in_disjunction, LogicalOperator::And);
+                for (const auto& expression_in_conjunction : inner_conjunction) {
+                  const auto evaluable_on_left_side =
+                      expression_evaluable_on_lqp(expression_in_conjunction, *join_node->left_input());
+                  const auto evaluable_on_right_side =
+                      expression_evaluable_on_lqp(expression_in_conjunction, *join_node->right_input());
+
+                  // We can only work with expressions that are specific to one side.
+                  if (evaluable_on_left_side && !evaluable_on_right_side && !aborted_left_side) {
+                    left_conjunction.emplace_back(expression_in_conjunction);
+                  }
+                  if (evaluable_on_right_side && !evaluable_on_left_side && !aborted_right_side) {
+                    right_conjunction.emplace_back(expression_in_conjunction);
+                  }
+                }
+
+                if (!left_conjunction.empty()) {
+                  // If we have found one or more predicates for the left side, connect them using AND and add them to
+                  // the disjunction that will be pushed to the left side.
+                  left_disjunction.emplace_back(inflate_logical_expressions(left_conjunction, LogicalOperator::And));
```

Comment:
Why do we 'and' them here? Can you direct me to an example/test here?

Reply:
If we have found one or more predicates for the left side, connect them using AND and add them to
the disjunction that will be pushed to the left side:
 Example: `(l1 AND r2) OR (l2 AND r1)` lead to `left_conjunction = [l1, l2]`, which is now turned
          into `l1 AND l2`.

Reply:
I don't see how that relates to `              // (l1 AND r2) OR (l2 AND r1) -> create (l1 OR l2) on left, (r1 OR r2) on right (example from above)`

Reply:
```
If we have found multiple predicates for the left side, connect them using AND and add them to
the disjunction that will be pushed to the left side:
 Example: `(l1 AND l2 AND r1) OR (l3 AND r2)` is first split into the two conjunctions. When
 looking at the first conjunction, l1 and l2 will end up in left_conjunction. Before it gets added
 to the left_disjunction, it needs to be connected using AND: (l1 AND l2).
 The result for the left_disjunction will be ((l1 AND l2) OR l3)